spec/rspec/rails/matchers/have_enqueued_mail_spec.rb

```diff
@@ -305,11 +313,6 @@ def test_email; end
     end
 
     it "generates a failure message with unmatching enqueued mail jobs" do
-      non_mailer_job = Class.new(ActiveJob::Base) do
-        def perform; end
-        def self.name; "NonMailerJob"; end
-      end
```

Comment:
(Since now there are two specs in this file (i.e. the spec being modified here, and the new spec that I added above) that have occasion to refer to a non-mailer job class, I am extracting this previously anonymous/local job class up to an actual class constant defined in the setup at the top of this file. The downside of this is that the global Ruby namespace in specs is now polluted by an additional constant, `NonMailerJob`, but personally I think that's worthwhile, for the upside of any specs that need to reference such a job being able to conveniently do so.)